### ISSUE TITLE
Winwheel.js segments and array

### ISSUE THREAD

**Issue by ladnur** [zarocknz/javascript-winwheel#9]
Hi, I first and foremost wanted to thank you for making this script and tutorial easy to comprehend and follow. I had one question about the segments: what I want to do with my wheel is to have each segment open up random pages that relates to each segment. So for example, I have 5 segments on my wheel, and I want my "lets play a game!" Segment to open up random html pages that have games on it, and I want my "lets eat!" Segment to open up random html pages that have recipes, etc. I would like to know how to assign each segment a certain array of pages, and have them correspond to each segment.

**Comment by ladnur** [comment]
`
<html >
    <head>

              <title>Poporro</title>
     <link rel="apple-touch-icon" sizes="57x57" href="apple-icon-57x57.png">
<link rel="apple-touch-icon" sizes="60x60" href="apple-icon-60x60.png">
<link rel="apple-touch-icon" sizes="72x72" href="apple-icon-72x72.png">
<link rel="apple-touch-icon" sizes="76x76" href="apple-icon-76x76.png">
<link rel="apple-touch-icon" sizes="114x114" href="apple-icon-114x114.png">
<link rel="apple-touch-icon" sizes="120x120" href="apple-icon-120x120.png">
<link rel="apple-touch-icon" sizes="144x144" href="apple-icon-144x144.png">
<link rel="apple-touch-icon" sizes="152x152" href="apple-icon-152x152.png">
<link rel="apple-touch-icon" sizes="180x180" href="apple-icon-180x180.png">
<link rel="icon" type="image/png" sizes="192x192"  href="android-icon-192x192.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="icon" type="image/png" sizes="96x96" href="favicon-96x96.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
        <script src='Winwheel.js'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/gsap/1.19.0/TweenMax.min.js"></script>
        <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
        
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
         
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
        
        <script src="dist/sweetalert.min.js"></script> 
        <link rel="stylesheet" type="text/css" href="dist/sweetalert.css">


<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap-theme.min.css" integrity="sha384-rHyoN1iRsVXV4nD0JutlnGaslCJuC7uwjduW9SVrLvRYooPp2bWYgmgJQIXwl/Sp" crossorigin="anonymous">
    

<link rel=”stylesheet” id=”font-awesome-css” href=”//netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css” type=”text/css” media=”screen”>
      
    </head>
    
    <style>
        
           html{
        background-color: #ffd57e;

    }
    
    body{
        background-color: #ffd57e; 
        
    }
    
    a{
        
        font-family: 'Antique Olive Std';
           color: white; 
        text-align: center;
        font-size: 20px

    }
    a:hover{
        color: black;
        text-decoration: none;
    }
    .border{
        position: fixed;
        border: 2px solid #DDB264;
        background-color: #DDB264;
        width: 150px;
        height: 170px;
        margin-top: 40px;
        padding-left: 30px;
        border-radius: 0px 20px 20px 0px;
        -webkit-box-shadow: 15px 15px 30px 0px rgba(0,0,0,0.09);
        z-index: 2;
      
    }
    #canvasContainer {
         margin: auto; left:0; right:0; top:0; bottom:0;
  position: relative;
  
    background-image: url(stand.png);
    background-repeat: no-repeat;   /* Ensure that background does not repeat */
    background-position: center;    /* Ensure image is centred */
    width: 400px;                   /* Width and height should at least be the same as the canvas */
    height: 500px;
            z-index: 1;
      
}

        
#canvas {
      margin: auto; left:0; right:0; top:0; bottom:0;
  position: absolute;
    z-index: 3;
        }

 
#prizePointer {
    margin: auto; left:0; right:0; top:0; bottom:0;
  position: absolute;
  text-align: center;
  margin-top: 100px;
    z-index: 4;

}
        
        #img1{
         margin: auto; left:0; right:0; top:0; bottom:0;
  position: absolute;
  text-align: center;
  margin-top: 289px; 
            z-index: 5;
        }
        #section1 {font-family: 'Antique Olive Std'; padding-top:50px;height:600px;color: #fff; background-color: #ffd57e;}
         #section2 {font-family: 'Antique Olive Std';padding-top:50px;height:600px;color: #fff; background-color: #ffd57e;}
  #section3 {font-family: 'Antique Olive Std';padding-top:50px;height:200px;color: #fff; background-color: #ffd57e;}
        
        .logo{
            position: absolute; margin-left: 500px; opacity: 0.5; overflow: hidden;
        }
    </style>
    <body>
        
    <body>
       

         <div class="border">
<br><a href="index.html"> Home </a>
<br><a href="wheel.html"> Spin </a> 
<br><a href="#section2"> About </a>
<br><a href="#section3"> Contact </a>
    </div>
        <center> <div id="section1" class="container-fluid">
  <h1>SPIN</h1>
 <img id="prizePointer" src="pointer.png" alt="V" />
         <img id="img1" src="spin.png" onClick="firstWheel.startAnimation();"> </img>
           <div id="canvasContainer">
              
        <center> <canvas id='myCanvas' width='400' height='418'> </center>

        </canvas>
           
           </div>
            </div> </center>
      <center>  <div id="section2" class="container-fluid">
  <h1>Section 2</h1>
          </div></center>
<center><div id="section3" class="container-fluid">
    <h1>Section 3</h1></div></center>
    <img class="logo" src="logo.png" ></img>
        
    </body>
        
      
   
       
        <script>
            
            
            
 var firstWheel = new Winwheel({
            'canvasId'     : 'myCanvas',
            'drawMode'     : 'image',
            'numSegments'  : 5,
            'segments'    :
       [
        {'text' : 'create some Origami'},
           {'text' : 'make weird things'},
           {'text' : 'cook and eat'},
           {'text' : 'learn a new skill'},
           {'text' : 'make something fun'}
        ],
            'imageOverlay' : true,
            'lineWidth'    : 0.0005,
            'strokeStyle'  : 'black',
            'animation' :                   // Note animation properties passed in constructor parameters.
            {
                'type'     : 'spinToStop',  // Type of animation. 
                'duration' : 5,             // How long the animation is to take in seconds.
                'spins'    : 8,              // The number of complete 360 degree rotations the wheel is to do.
                // Remember to do something after the animation has finished specify callback function.
            'callbackFinished' : 'alertPrize()'
            }
        });
      
            
             // This function called after the spin animation has stopped.
    function alertPrize()
    {
        // Call getIndicatedSegment() function to return pointer to the segment pointed to on wheel.
        var winningSegment = firstWheel.getIndicatedSegment();
         
        // Basic alert of the segment text which is the prize name.
        alert("Let's " + winningSegment.text + "!");
         window.location= 'index.html';
       
         }
        
        // Create new image object in memory.
        var firstImg = new Image();
        
        // Create callback to execute once the image has finished loading.
        firstImg.onload = function()
        {
            firstWheel.wheelImage = firstImg;    // Make wheelImage equal the loaded image object.
            firstWheel.draw();                    // Also call draw function to render the wheel.
        };
        
        // Set the image source, once complete this will trigger the onLoad callback (above).
        firstImg.src = "wheel.png";
         
        
        </script>
        
        
        
        
    </body>
</html>
`

**Comment by zarocknz** [comment]
Hi @ladnur, thanks for getting in touch.

I am glad you are finding the Winwheel.js useful and that you can follow the tutorials. Interesting to hear your idea about the wheel opening up random pages, sounds fun.

Yes, what you are wanting to is easily possible. The Winwheel.js code has a feature where you can specify any properties you like for the segments, including ones that are not built in. Just add whatever properties you like to the configuration passed in to the Winwheel constructor.

This allows an array of URLs to be added to each segment and then a random url for the winning segment to be selected and the window's location changed to it.

The first step is to define the urls for each segment, in the config information like this...

``` javascript
'segments' :
[
    {
        'fillStyle' : '#eae56f', 
        'text' : 'Prize 1',
        'urls' : 
        [
            'http://www.dougtesting.net', 
            'http://www.google.com', 
            'http://www.wikipedia.com'
        ]
    },
    {
        'fillStyle' : '#89f26e', 
        'text' : 'Prize 2',
        'urls' : 
        [
            'http://www.linkedin.com', 
            'http://www.twitter.com', 
            'http://www.facebook.com'
        ]
    },
    {
        'fillStyle' : '#7de6ef', 
        'text' : 'Prize 3',
        'urls' : 
        [
            'http://www.linkedin.com', 
            'http://www.twitter.com', 
            'http://www.facebook.com'
        ]
    }
],
```

Then the next step is to change the alertPrize() function called when the spinning has stopped to work out a random url for the winning segment and then change the location of the page to that url...

``` javascript
function alertPrize()
{
    // Get the segment indicated by the pointer on the wheel.
    var winningSegment = theWheel.getIndicatedSegment();

    // Get a random url for this segment.
    var urlNumber = Math.floor(Math.random() * winningSegment.urls.length);

    // Change the window's location to it.
    window.location = winningSegment.urls[urlNumber];    
}
```

That's it.

Let me know how you get on, or if you have further questions.

Kind regards,
DouG.

**Comment by ladnur** [comment]
Thank you! I used it to build my website poporro.com, and it works perfectly. The code was so easy I'm disappointed I didn't think of it before I asked you lol. Thanks again!

**Comment by vcam100** [comment]
hi , please help me . i want to use a variable as the segments.  now i made a string var that prints out like this : 
[{'text' : 'rrr'},{'text' : 'ooo'},{'text' : 'pppp'},{'text' : 'qqqq'},{'text' : 'wwww'},{'text' : 'tttt'},{'text' : 'uuuu'}]

how do i pass this one to the 'segments' : .....    ?

i have tried  just writing the variable and a comma after but it does not work. 

   'numSegments'     : secs,         // Specify number of segments.
                'segments'        : result,           // Define segments 
                'animation' :           // Specify the animation to use.

result is the string from above , and 
btw , secs , is a number and that one worked. 

Thank you very much  for everything !
